Descent3/menu.cpp

```diff
@@ -1090,25 +1073,23 @@ static inline int count_missions(const std::vector<std::filesystem::path> &missi
   return c;
 }
 
-static inline int generate_mission_listbox(newuiListBox *lb, int n_maxfiles, char **filelist,
+static inline int generate_mission_listbox(newuiListBox *lb, std::vector<std::filesystem::path> &filelist,
                                            const std::vector<std::filesystem::path> &missions_directories) {
   int c = 0;
   for (const auto &missions_directory : missions_directories) {
     ddio_DoForeachFile(
-        missions_directory, std::regex(".*\\.mn3"), [&c, &lb, &n_maxfiles, &filelist](const std::filesystem::path &path) {
+        missions_directory, std::regex(".*\\.mn3"), [&c, &lb, &filelist](const std::filesystem::path &path) {
           tMissionInfo msninfo{};
-          if (c < n_maxfiles) {
-            if (stricmp(path.filename().u8string().c_str(), "d3_2.mn3") == 0)
-              return;
-            if (GetMissionInfo(path.filename().u8string().c_str(), &msninfo) && msninfo.name[0] && msninfo.single) {
-              filelist[c] = mem_strdup(path.filename().u8string().c_str());
-              lb->AddItem(msninfo.name);
-              c++;
-              if (!(c % 2))
-                DoWaitMessage(true);
-            }
-          }
-        });
+          if (stricmp(path.filename().u8string().c_str(), "d3_2.mn3") == 0)
+          return;
+        if (GetMissionInfo(path.filename(), &msninfo) && msninfo.name[0] && msninfo.single) {
+          filelist.push_back(path.filename());
+          lb->AddItem(msninfo.name);
+          c++;
+          if (!(c % 2))
```

Comment:
I know this code was there before, I don't see why do we display wait message only for even `c`

Reply:
Just saving CPU cycles on fetching mission list.